src/main/java/tracko/logic/parser/order/FindOrderCommandParser.java

```diff
@@ -1,32 +1,106 @@
+import static tracko.logic.parser.CliSyntax.FLAG_UNPAID;
+import static tracko.logic.parser.CliSyntax.PREFIX_ADDRESS;
+import static tracko.logic.parser.CliSyntax.PREFIX_ITEM;
+import static tracko.logic.parser.CliSyntax.PREFIX_NAME;
 
 import java.util.Arrays;
+import java.util.stream.Stream;
 
 import tracko.logic.commands.order.FindOrderCommand;
+import tracko.logic.parser.ArgumentMultimap;
+import tracko.logic.parser.ArgumentTokenizer;
+import tracko.logic.parser.Flag;
+import tracko.logic.parser.FlagTokenizer;
 import tracko.logic.parser.Parser;
+import tracko.logic.parser.Prefix;
 import tracko.logic.parser.exceptions.ParseException;
-import tracko.model.order.OrderContainsKeywordsPredicate;
+import tracko.model.order.OrderMatchesFlagsAndPrefixPredicate;
 
 /**
  * Parses input arguments and creates a new FindOrderCommand object
  */
 public class FindOrderCommandParser implements Parser<FindOrderCommand> {
+    private static final String DELIMITER = "\\s+";
+
     /**
      * Parses the given {@code String} of arguments in the context of the FindOrderCommand
      * and returns a FindOrderCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
-    public FindOrderCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindOrderCommand.MESSAGE_USAGE));
+    public FindOrderCommand parse(String userInput) throws ParseException {
+        ArgumentMultimap prefixMultimap = ArgumentTokenizer.tokenize(userInput, PREFIX_NAME, PREFIX_ADDRESS,
+                PREFIX_ITEM);
+
+        ArgumentMultimap flagMultimap = FlagTokenizer.tokenize(userInput, FLAG_PAID, FLAG_UNPAID,
+                FLAG_DELIVERED, FLAG_UNDELIVERED);
+
+        boolean isAnyPrefixPresent = isAnyPrefixPresent(prefixMultimap, PREFIX_NAME, PREFIX_ADDRESS,
+                PREFIX_ITEM);
+        boolean isAnyFlagPresent = isAnyFlagPresent(flagMultimap, FLAG_PAID, FLAG_UNPAID,
+                FLAG_DELIVERED, FLAG_UNDELIVERED);
+        boolean isPrefixMultimapPreambleEmpty = prefixMultimap.getPreamble().isEmpty();
+        boolean isFlagMultimapPreambleEmpty = flagMultimap.getPreamble().isEmpty();
+
+        if (!(isAnyPrefixPresent || isAnyFlagPresent)
+                || !(isFlagMultimapPreambleEmpty || isPrefixMultimapPreambleEmpty)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindOrderCommand.MESSAGE_USAGE));
+        }
+
+        String nameKeywordsString = prefixMultimap.getValue(PREFIX_NAME).orElse("").trim();
+        String addressKeywordsString = prefixMultimap.getValue(PREFIX_ADDRESS).orElse("").trim();
+        String itemKeywordsString = prefixMultimap.getValue(PREFIX_ITEM).orElse("").trim();
+
+        String [] nameKeywords = splitKeywords(nameKeywordsString);
```

Comment:
You might want to consider removing the whitespace between `String` and `[]` here, for code quality